code/game/machinery/computer/robot.dm

```diff
@@ -76,6 +90,23 @@
 		)
 		data["drones"] += list(drone_data)
 
+
+	data["uploads"] = list()
+	for(var/obj/machinery/computer/upload/U in GLOB.uploads_list)
+		if(stat & (NOPOWER|BROKEN))
+			continue
+		if(get_virtual_z_level() != (get_turf(U)).get_virtual_z_level())
```

Comment:
What if (in the far future) we have multi-z maps? In addition to checking virtual z levels, also check `is_station_level()`, since different station z levels will have different virtual levels assigned, but uploads on these should still be tracked.